lib/tlscontext.c

```diff
@@ -513,20 +516,84 @@ tls_context_setup_dh(TLSContext *self)
   return ctx_dh_success;
 }
 
+static gboolean
+tls_context_load_pkcs12(TLSContext *self)
+{
+  if (!file_exists(self->pkcs12_file))
+    return FALSE;
+
+  FILE *p12_file = fopen(self->pkcs12_file, "rb");
+
+  if (!p12_file)
+    return FALSE;
+
+  PKCS12 *pkcs12 = d2i_PKCS12_fp(p12_file, NULL);
```

Comment:
Could you factor out a `static PKCS12 *_load_pkcs12_file(const gchar *filename);` function?

Reply:
Done.